engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityCache.java

```diff
@@ -0,0 +1,350 @@
+            return create(prefab, position);
+        }
+        return create();
+    }
+
+    @Override
+    public EntityRef create(Prefab prefab, Vector3f position) {
+        List<Component> components = Lists.newArrayList();
+        for (Component component : prefab.iterateComponents()) {
+            Component newComp = entityManager.getComponentLibrary().copy(component);
+            components.add(newComp);
+            if (newComp instanceof LocationComponent) {
+                LocationComponent loc = (LocationComponent) newComp;
+                loc.setWorldPosition(position);
+            }
+        }
+        components.add(new EntityInfoComponent(prefab, prefab.isPersisted(), prefab.isAlwaysRelevant()));
+        return create(components);
+    }
+
+    @Override
+    public EntityRef create(Prefab prefab) {
+        List<Component> components = Lists.newArrayList();
+        for (Component component : prefab.iterateComponents()) {
+            components.add(entityManager.getComponentLibrary().copy(component));
+        }
+        components.add(new EntityInfoComponent(prefab, prefab.isPersisted(), prefab.isAlwaysRelevant()));
+        return create(components);
+    }
+
+    @Override
+    public EntityRef create(Prefab prefab, Vector3f position, Quat4f rotation) {
+        List<Component> components = Lists.newArrayList();
+        for (Component component : prefab.iterateComponents()) {
+            Component newComp = entityManager.getComponentLibrary().copy(component);
+            components.add(newComp);
+            if (newComp instanceof LocationComponent) {
+                LocationComponent loc = (LocationComponent) newComp;
+                loc.setWorldPosition(position);
+                loc.setWorldRotation(rotation);
+            }
+        }
+        components.add(new EntityInfoComponent(prefab, prefab.isPersisted(), prefab.isAlwaysRelevant()));
+        return create(components);
+    }
+
+
+    /**
+     * Destroys this entity, sending event
+     *
+     * @param entityId
+     */
+    @Override
+    public void destroy(long entityId) {
+        // Don't allow the destruction of unloaded entities.
+        if (!entityManager.idLoaded(entityId)) {
+            return;
+        }
+        EntityRef ref = createEntityRef(entityId);
+        if (entityManager.getEventSystem() != null) {
+            entityManager.getEventSystem().send(ref, BeforeDeactivateComponent.newInstance());
+            entityManager.getEventSystem().send(ref, BeforeRemoveComponent.newInstance());
+        }
+        entityManager.notifyComponentRemovalAndEntityDestruction(entityId, ref);
+        destroy(ref);
+    }
+
+    private void destroy(EntityRef ref) {
+        // Don't allow the destruction of unloaded entities.
+        long entityId = ref.getId();
+        entityStore.remove(entityId);
+        entityManager.remove(entityId);
+        if (ref instanceof PojoEntityRef) {
```

Comment:
I don't see why you shouldn't be able to invalidate any EntityRef, even if that doesn't do anything for some implementations of it. I've added that in 15ca85c; I think that getting rid of the `instanceof` makes it a lot cleaner.